src/buffer.js

```diff
@@ -0,0 +1,127 @@
+const {contextRecordKey, deferred} = require('./utils');
+
+function queryBuffer(config, emitter, targetStore) {
+  const buffers = [];
+
+  let numCoalesced = 0;
+  let numCached = 0;
+  let numMisses = 0;
+
+  class RequestBuffer {
+    constructor(key, params) {
+      this.uid = Math.random().toString(36);
+      this.state = 0;
```

Comment:
[⚪] What does this map to? Any chance we can replace those magic values for an "enum" ?

Reply:
Good point, I'll make one

Reply:
@mats852 After switching to an enum, memory usage dropped by 4x ! 
Now that's an interesting case study!

Reply:
Never expected this! In hindsight it can make sense. Great!!